### ISSUE TITLE
(RES for Chrome) The order of the fIlters in filterline is not maintained (they are sorted alphabetically)

### ISSUE THREAD

**Issue by darkred** [honestbleeps/Reddit-Enhancement-Suite#4148]
- My browser: Chrome 57 in win10
- My RES version: 5.4.3
- Am I in reddit beta (yes/no): no

Order of the fIlters in filterline is not maintained after you close a reddit.com tab

STR
- open https://www.reddit.com/r/CrackWatch/
- add these flair filters: a `Humor`, `Repack`, `Discussion`, `NFO`
Notice that their order is like this:
![2017-03-30_151745](https://cloud.githubusercontent.com/assets/723651/24503786/3f5a4468-155c-11e7-93bf-a936abf59d29.jpg)

- Now close the reddit.com tab, open another tab and load reddit.com. Here is how the filters are displayed now:
![2017-03-30_151802](https://cloud.githubusercontent.com/assets/723651/24503792/45402c8a-155c-11e7-9711-da2a7733643d.jpg)

The issue doesn't occur in Firefox 52 with RES 5.4.3

**Comment by andytuba** [comment]
Could you make this kind of post to /r/RESissues in the future? We prefer restricting GitHub issues to todo tasks.

---

Does the order of filters matter for your user experience, or are you just noting the inconsistency between the browsers?

**Comment by darkred** [comment]
> Could you make this kind of post to /r/RESissues in the future? We prefer restricting GitHub issues to todo tasks.

Sure (and sorry, I didn't know that)

> Does the order of filters matter for your user experience, or are you just noting the inconsistency between the browsers?

Yes, the order matters for me because I have my filters in Firefox like this:
![2017-03-30_225312](https://cloud.githubusercontent.com/assets/723651/24523510/e6f341a4-159b-11e7-87ec-798eaa470690.jpg)
because I **never** disable the filters `Humor` and `Repack`,
but I frequently toggle the `Discussion` and `NFO` ones.

**Comment by andytuba** [comment]
>> Could you make this kind of post to /r/RESissues in the future? We prefer restricting GitHub issues to todo tasks.
> Sure (and sorry, I didn't know that)

Maybe I should improve the issue template message to not just 'Is RES broken?' but 'Is some feature of RES not working for you?'

> I never disable the filters...

Sound like you should remove those filters from Filterline and instead add them to [FilteReddit's **flair** filters](https://reddit.com/r/crackwatch/#res:settings/filteReddit/flair).  While you're in there, you might want to set up a "Crackwatch chat" [Custom Toggle](https://www.reddit.com/r/CrackWatch/#res:settings/customToggles) (put that text in both the `name` and `menuItem` text fields, so the toggle shows up in the RES gear menu in the top right) to control a [Custom Filter](https://www.reddit.com/r/CrackWatch/#res:settings/filteReddit/customFilters) for

    all of:
        Subreddit: Crackwatch
       any of:
           post flair: Discussion
           post flair: NFO
      custom toggle: Crackwatch chat

**Comment by darkred** [comment]
Thank you for the suggestions.
So, I tried your instructions, but unfortunately it filters all posts in r/CrackWatch, (=no post appears in any pages) , independently of if I have it enabled or not:
![2017-03-30_235459](https://cloud.githubusercontent.com/assets/723651/24525796/67aa3110-15a4-11e7-9493-a5e4803b6fe4.jpg)
![2017-03-30_235512](https://cloud.githubusercontent.com/assets/723651/24525797/67aaf8fc-15a4-11e7-97ee-6ecee4076c3b.jpg)



And in any case, I'd prefer to use the Filterline instead of the flair filters, because I find it easier to use.
And maybe I wasn't accurate but, when saying "I never disable the filters", it wasn't meant literally, it's more "almost never/rarely"

**Comment by darkred** [comment]
Thank you for the suggestions.
So, I tried your instructions, but unfortunately (if I entered them correctly) it filters all posts in `r/CrackWatch` (=no post appears in any page) , independently of if I have it enabled or not:
![2017-03-30_235459](https://cloud.githubusercontent.com/assets/723651/24525796/67aa3110-15a4-11e7-9493-a5e4803b6fe4.jpg)
![2017-03-30_235512](https://cloud.githubusercontent.com/assets/723651/24525797/67aaf8fc-15a4-11e7-97ee-6ecee4076c3b.jpg)



And in any case, I'd prefer to use the Filterline instead of the flair filters, 
because I find it much easier to use.
And maybe I wasn't accurate but, when saying "I never disable those filters", it wasn't meant literally, it's more "almost never/rarely". In fact the order I have for those filters is more like "from "the least used" --> "the most used"
